### ISSUE TITLE
Duplicate images in mobile app when posted using 'share' menu from another source

### ISSUE THREAD

**Issue by gigawhitlocks** [mattermost/mattermost-server#22206]
Per Mattermost guidelines, GitHub issues are for bug reports: <https://handbook.mattermost.com/contributors/contributors/ways-to-contribute/>.

For troubleshooting see: https://forum.mattermost.com/.
For feature proposals see: https://portal.productboard.com/mattermost/33-what-matters-to-you

If you've found a bug--something appears unintentional--please follow these steps:

1. Confirm you’re filing a new issue. [Search existing tickets in Jira](https://mattermost.atlassian.net/jira/software/c/projects/MM/issues/) to ensure that the ticket does not already exist.
2. Confirm your issue does not involve security. Otherwise, please see our [Responsible Disclosure Policy](https://mattermost.com/security-vulnerability-report/).
3. [File a new issue](https://github.com/mattermost/mattermost-server/issues/new) using the format below. Mattermost will confirm steps to reproduce and file in Jira, or ask for more details if there is trouble reproducing it. If there's already an existing bug in Jira, it will be linked back to the GitHub issue so you can track when it gets fixed. 

#### Summary
When using the 'share' option to post an image from another application into Mattermost on Android in the Mobile v2 app, images are double-posted in the channel due to the inclusion of the URL of the image as well as the image itself in the post.

The URL is included as message text and is expanded, and then the actual data of the image is included underneath it, causing the image to render twice in the channel.

This problem existed on v1 as well, but v1 gave you the option to delete the URL before posting. This was annoying, but you could avoid the double-post. Now, you must share the image to Mattermost, then switch apps to Mattermost, edit the post, and delete the URL so that the image only appears once.

#### Steps to reproduce
Open another app, such as Firefox Mobile, and find an image. Long-press on the image and choose "share" and then choose "Mattermost". Share the image to a channel without adding any additional message. The image will appear in the channel first as the preview of the URL that is included, and then again as an attachment.

#### Expected behavior
The image appears only once as an attachment

#### Observed behavior (that appears unintentional) 
The image appears twice

#### Possible fixes
Just don't post the URL of the shared attachment in addition to the attachment when using the 'Share' menu. If the user wants the URL to be posted as well, the user can copy and paste it into the optional comment field while posting.
